tests/Platforms/MySQLPlatformTest.php

```diff
@@ -27,4 +28,26 @@ public function testDropIndexSQLRequiresTable(): void
         $this->expectException(InvalidArgumentException::class);
         $this->platform->getDropIndexSQL('foo');
     }
+
+    public function testCollationOptionIsTakenIntoAccount(): void
+    {
+        $table = new Table('quotations');
+        $table->addColumn('id', 'integer');
+        $table->addOption('collation', 'my_collation');
+        self::assertStringContainsString(
+            'my_collation',
+            $this->platform->getCreateTableSQL($table)[0]
+        );
+    }
+
+    public function testCollateOptionIsStillSupported(): void
+    {
+        $table = new Table('quotations');
+        $table->addColumn('id', 'integer');
+        $table->addOption('collate', 'my_collation');
+        self::assertStringContainsString(
+            'my_collation',
+            $this->platform->getCreateTableSQL($table)[0]
+        );
+    }
```

Comment:
Although both tests look a lot like each other, I elected not to use a data provider here, as the second test should diverge in 3.4 because of the deprecation.